test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/JacocoConfig.java

```diff
@@ -11,10 +11,11 @@
 public class JacocoConfig {
 
     /**
-     * The jacoco data file
+     * The jacoco data file. By default this will be target/jacoco-quarkus.exec.
+     * The path can be relative (to the module) or absolute.
      */
-    @ConfigItem(defaultValue = "jacoco-quarkus.exec")
-    public String dataFile;
+    @ConfigItem
+    public Optional<String> dataFile;
```

Comment:
What's the rationale of this change?

Reply:
I changed the default to be the equivalent of `target/jacoco-quarkus.exec`, but since I wanted to reuse `outputTargetBuildItem.getOutputDirectory()` and `module.getWorkspaceModule().getBuildDir()` in the processor, I could not use a `WithDefault` anymore.
so now it is optional, but a default is calculated in the processor directly.
there are no changes for people who did not set `data-file`.
for people that did set `data-file=foo`, they need to change it to `target/foo`.
your advice more than welcome.

Reply:
Makes sense, thanks.

What I would like however is some Javadoc explaining this process so users who might want to configure it can find this information on our config page (or their IDE if it supports it)

Reply:
I guess it's fine.